lib/Intlc/Backend/TypeScript/Compiler.hs

```diff
@@ -25,8 +25,14 @@ compileNamedExport x l k v =
 compileTypeof :: InterpStrat -> ICU.Message -> Text
 compileTypeof x = let o = fromStrat x in flip runReader o . typeof . fromMsg o
 
-reactImport :: Text
-reactImport = "import React, { ReactElement } from 'react'"
+isTypeScriptReact :: Translation -> Bool
+isTypeScriptReact (Translation _m  be) = case be of
+  TypeScript      -> False
+  TypeScriptReact -> True
+
+
+buildReactImport :: Dataset Translation -> Maybe Text
+buildReactImport = fmap (const "import React, { ReactElement } from 'react'") . find isTypeScriptReact
```

Comment:
I'd argue this belongs in the JS compiler, which handles the runtime stuff for TS.